pkg/distro/id.go

```diff
@@ -22,6 +24,16 @@ func (id ID) String() string {
 	return fmt.Sprintf("%s-%d.%d", id.Name, id.MajorVersion, id.MinorVersion)
 }
 
+func (id ID) Version() (*version.Version, error) {
+	var verStr string
+	if id.MinorVersion == -1 {
+		verStr = fmt.Sprintf("%d", id.MajorVersion)
+	} else {
+		verStr = fmt.Sprintf("%d.%d", id.MajorVersion, id.MinorVersion)
+	}
```

Comment:
Could we split this out into a separate `VersionString()` and then also re-use it in `String()`? They currently duplicated the `%d` vs `%d.%d` logic.